Sources/Datadog/RUM/Instrumentation/Views/SwiftUI/SwiftUIViewModifier.swift

```diff
@@ -0,0 +1,60 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+#if canImport(SwiftUI)
+import SwiftUI
+
+/// `SwiftUI.ViewModifier` for RUM which invoke `startView` and `stopView` from the
+/// global RUM Monitor when the modified view appears and disappears.
+@available(iOS 13, *)
+internal struct RUMViewModifier: SwiftUI.ViewModifier {
+    /// The Content View identifier.
+    /// The id will be unique per modified view.
+    let id: String = UUID().uuidString
+
+    /// View Name used for RUM Explorer.
+    let name: String
+
+    /// View Path used for RUM Explorer.
+    let path: String
+
+    /// Custom attributes to attach to the View.
+    let attributes: [AttributeKey: AttributeValue]
+
+    func body(content: Content) -> some View {
+        content.onAppear {
+            RUMInstrumentation.instance?.onAppear(
+                identity: id,
+                name: name,
+                path: path,
+                attributes: attributes
+            )
+        }
+        .onDisappear {
+            RUMInstrumentation.instance?.onDisappear(identity: id)
+        }
+    }
+}
+
+@available(iOS 13, *)
+public extension SwiftUI.View {
+    /// Monitor this view with Datadog RUM. A start and stop events will be logged when this view appears
+    /// and disappears.
+    ///
+    /// - Parameters:
+    ///   - name: the View name used for RUM Explorer.
+    ///   - attributes: custom attributes to attach to the View.
+    /// - Returns: This view after applying a `ViewModifier` for monitoring the view.
+    func trackRUMView(
+        name: String,
+        attributes: [AttributeKey: AttributeValue] = [:]
+    ) -> some View {
+        let path = "\(name)/\(typeDescription.hashValue)"
+        return modifier(RUMViewModifier(name: name, path: path, attributes: attributes))
+    }
```

Comment:
We cannot call .`onAppear()/ .onDisappear()` form a `SwiftUI.View`, a view only declare its body. That's why I left it out of unit tests and cover this in integration tests. I could try to workaround that by embedding the view in a `UIHostController` and see if I can trigger the events from it.

Reply:
If it's not too complex, IMO we can give it a shot. Otherwise, integration tests coverage is fine 👌 and definitely better than tricky or flaky units.

Reply:
I've added a unit test using `UIHostingController`. It also involve a mocked `SwiftUIViewHandler` and a new `RUMInstrumentation.init`.
`onAppear` and `onDisappear` have to be tested in a single unit because it has to wait for appearance before disappearing.